pyiron_atomistics/dft/job/bader.py

```diff
@@ -0,0 +1,124 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import numpy as np
+import os
+import subprocess
+
+__author__ = "Sudarsan Surendralal"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Sudarsan Surendralal"
+__email__ = "surendralal@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+
+class Bader:
+    """
+    Module to apply the Bader charge partitioning scheme to finished DFT jobs. This module is interfaced with the
+    `Bader code`_ from the Greame Henkelmann group.
+
+    .. _Bader code: http://theory.cm.utexas.edu/henkelman/code/bader
+    """
+    def __init__(self, job):
+        """
+        Initialize the Bader module
+
+        Args:
+            job (pyiron_atomistics.dft.job.generic.GenericDFTJob): A DFT job instance (finished/converged job)
+        """
+        self.job = job
+        self._working_directory = job.working_directory
+        self._structure = job.structure
+
+    def create_cube_files(self):
+        """
+        Create CUBE format files of the total and valce charges to be used by the Bader program
+        """
+        cd_val, cd_total = self.job.get_valence_and_total_charge_density()
+        cd_val.write_cube_file(filename=os.path.join(self._working_directory, "valence_charge.CUBE"))
+        cd_total.write_cube_file(filename=os.path.join(self._working_directory, "total_charge.CUBE"))
+
+    def call_bader_from_job(self, extra_arguments=None):
```

Comment:
I thought about this. But since this module is technical a "job", I consciously avoided `run` or `calculate`. Perhaps `compute_bader_charges` is a more suitable name?

Reply:
Yeah, I don't object to that name. IMO even `compute` is sufficient given the context that it is the one public method of the `Bader` class. 

> But since this module is technical a "job"...

I'm still confused by this. As per my main review comment, isn't it explicitly *not* a job? In this case I also see a good justification for avoiding `run` (job's have `run` and this isn't a job). But I still understand it to be not-a-job, and don't think we need to specially avoid `calculate`, `compute`, or similar.

Reply:
> I'm still confused by this. As per my main review comment, isn't it explicitly _not_ a job? In this case I also see a good justification for avoiding `run` (job's have `run` and this isn't a job). But I still understand it to be not-a-job, and don't think we need to specially avoid `calculate`, `compute`, or similar.

Sorry my bad, I meant "technically **not** a job" in my previous comment. I shouldn't post comments after 8 PM

Reply:
> I shouldn't post comments after 8 PM

😆 fair! I can relate!